### DISCUSSION TITLE
advanced_test这样报错，是什么原因？

### DISCUSSION THREAD

**Discussion by lhyyy1** [SUSTech-CS305-Fall22/CS305-Project-Skeleton#75] [Q&A]
如图所示
![image](https://user-images.githubusercontent.com/120730404/211955896-58fdc48e-e26f-460f-901f-60e28357d5aa.png)

**Comment by lhyyy1** [comment]
似乎是因为result这个目录已经存在就会报错？

**Reply by TANSixu** [reply]
好怪，rmtree理论上应该已经把它删掉了。请问这个问题是每次都会出现嘛？

**Reply by lhyyy1** [reply]
是的，而且我把mkdir这一行给注释掉就能正常跑

**Reply by TANSixu** [reply]
> 是的，而且我把mkdir这一行给注释掉就能正常跑

诶，那意思是results文件夹没有被rmtree删除嘛？

**Reply by lhyyy1** [reply]
> > 是的，而且我把mkdir这一行给注释掉就能正常跑
> 
> 诶，那意思是results文件夹没有被rmtree删除嘛？

跑的过程中results文件夹经历了一个被清空然后被放入结果的过程，所以我猜测rmtree只把内部清空了而没有删掉这个目录（

**Reply by TANSixu** [reply]
> > > 是的，而且我把mkdir这一行给注释掉就能正常跑
> > 
> > 
> > 诶，那意思是results文件夹没有被rmtree删除嘛？
> 
> 跑的过程中results文件夹经历了一个被清空然后被放入结果的过程，所以我猜测rmtree只把内部清空了而没有删掉这个目录（

应该是rmtree出了问题，请将上面的rmtree ignore_errors改成false在试一次呢，并且反馈一下报错信息

**Reply by lhyyy1** [reply]
________________________ ERROR at setup of test_finish _________________________

    @pytest.fixture(scope='module')
    def advance_session():
        success = False
        time_max = 640
        if os.path.exists("test/tmp5/results"):
>           shutil.rmtree("test/tmp5/results", ignore_errors=False)

/home/lhyyy/Share/test/advance_test.py:71: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3.8/shutil.py:722: in rmtree
    onerror(os.rmdir, path, sys.exc_info())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

path = 'test/tmp5/results', ignore_errors = False
onerror = <function rmtree.<locals>.onerror at 0x7f5c20d48700>

    def rmtree(path, ignore_errors=False, onerror=None):
        """Recursively delete a directory tree.
    
        If ignore_errors is set, errors are ignored; otherwise, if onerror
        is set, it is called to handle the error with arguments (func,
        path, exc_info) where func is platform and implementation dependent;
        path is the argument to that function that caused it to fail; and
        exc_info is a tuple returned by sys.exc_info().  If ignore_errors
        is false and onerror is None, an exception is raised.
    
        """
        sys.audit("shutil.rmtree", path)
        if ignore_errors:
            def onerror(*args):
                pass
        elif onerror is None:
            def onerror(*args):
                raise
        if _use_fd_functions:
            # While the unsafe rmtree works fine on bytes, the fd based does not.
            if isinstance(path, bytes):
                path = os.fsdecode(path)
            # Note: To guard against symlink races, we use the standard
            # lstat()/open()/fstat() trick.
            try:
                orig_st = os.lstat(path)
            except Exception:
                onerror(os.lstat, path, sys.exc_info())
                return
            try:
                fd = os.open(path, os.O_RDONLY)
            except Exception:
                onerror(os.open, path, sys.exc_info())
                return
            try:
                if os.path.samestat(orig_st, os.fstat(fd)):
                    _rmtree_safe_fd(fd, path, onerror)
                    try:
>                       os.rmdir(path)
E                       OSError: [Errno 26] Text file busy: 'test/tmp5/results'

/usr/lib/python3.8/shutil.py:720: OSError

**Reply by TANSixu** [reply]
请问是否是使用的虚拟机和主机共享文件夹的方式开发的？我查找到有相关案例，如果在win上打开了这个文件夹或者文件夹中的内容会引起rmtree出错，需要关闭主机中打开的

**Reply by lhyyy1** [reply]
确实是用这种方式，原来如此

**Reply by lhyyy1** [reply]
但是我把文件夹和IDE都关掉了还是这样报错（
